resources/type-system/inference.md

```diff
@@ -969,6 +969,23 @@ with respect to `L` under constraints `C0`
   - If for `i` in `0...m`, `Mi` is a subtype match for `Ni` with respect to `L`
   under constraints `Ci`.
 
+## Bound resolution
+
+For any type `T`, the _bound resolution_ of `T` is a type `U`, defined by the
+following recursive process:
+
+- If `T` is a type variable `X`, then let `B` be the bound of `X`. Then let `U`
+  be the bound resolution of `B`.
+
+- Otherwise, if `T` is a promoted type variable `X&B`, then let `U` be the bound
+  resolution of `B`.
+
+- Otherwise, let `U` be `T`.
+
+_Note that the spec notions of __dynamic__ boundedness and __Function__
+boundedness can be defined in terms of bound resolution, as follows: a type is
+__dynamic__ bounded iff its bound resolution is __dynamic__, and a type is
```

Comment:
(Not sure using bold for types is a good idea, when we also use bold for semantic functions like **UP**. Consider "is _dynamic bounded_ if and only if its bound resolution is the type `dynamic`" and similar for function.

Reply:
Yeah, I see what you mean. I’m usually in the habit of using a monospaced font for type names, but in this case, since I’m referring directly to text in the spec, I figured I would imitate the style of the spec. If you look at page 144 of https://spec.dart.dev/DartLangSpecDraft.pdf, it uses boldface for __dynamic__ and __Function__.